src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java

```diff
@@ -184,15 +184,70 @@ public class MongoSourceConfig extends AbstractConfig {
+          + "\nThe name in each pair is a namespace pattern, the value is a string representing the corresponding topic name template."
+          + " Pairs are ordered. When there are multiple pairs with equal namespace patterns (duplicates),"
+          + " they are deduplicated and only one pair is taken into account:"
+          + " its position is taken from the first pair among duplicates,"
+          + " its topic name template is taken from the last pair among duplicates."
+          + " After deduplication, pairs with an empty topic name template are ignored when computing topic names."
+          + " Note that a topic name computed based on this configuration is then decorated"
+          + " using the `topic.prefix` and `topic.suffix` configuration properties."
+          + "\nThere are two kinds of pairs:"
+          + "\n- Simple. The namespace pattern must not contain solidus '/' and can be either of the following:"
+          + "\n  - A namespace with a collection name, in which case it matches only that namespace."
+          + " The topic name template is interpreted as the topic name."
+          + "\n  - A namespace with only a database name, in which case it matches any namespace having that database name."
+          + " The matching namespace may either have a collection name, or not:"
+          + "\n    - If there is a collection name, then the topic name is computed"
+          + " by concatenating the topic name template and the collection name from the matching namespace, separated by `topic.separator`."
+          + "\n    - If there is no collection name, then the the topic name template is interpreted as the topic name."
+          + "\n- Regex. The namespace pattern starts with solidus '/',"
+          + " followed by a regular expression with the syntax and behavior as per `java.util.regex.Pattern`."
+          + " The topic name is computed by doing variable expansion on the topic name template."
+          + " The following variables are supported:"
+          + "\n  - `db` The database name from the matching namespace."
+          + "\n  - `sep` The value of the `topic.separator` configuration property."
+          + "\n  - `coll` The collection name from the matching namespace, or an empty string if there is no collection name."
+          + "\n  - `sep_coll` The value of the `coll` variable"
+          + " prefixed with the value of `sep` if and only if the value of `coll` is not empty."
+          + "\n  - `coll_sep` The value of the `coll` variable"
+          + " suffixed with the value of `sep` if and only if the value of `coll` is not empty."
+          + "\n  - `sep_coll_sep` The value of the `coll` variable"
+          + " prefixed and suffixed with the value of `sep` if and only if the value of `coll` is not empty."
+          + "\n To be expanded, a variable must be enclosed between curly brackets '{' and '}', for example '{db}'."
```

Comment:
The variable expansion syntax was chosen to be similar to the one proposed in [JEP 430: String Templates (Preview)](https://openjdk.org/jeps/430) with the difference that we don't use `\`.